src/odin_devices/dac63004.py

```diff
@@ -0,0 +1,392 @@
+            write = "0000" + ("000" * (3-index)) + voltagePowerDown + "0" + ("000" * index)
+            self.read_modify_write("COMMON_CONFIG", int(mask, 2), int(write, 2))
+        else:
+            print("Not a valid index.")
+            return
+        
+    def set_dac_as_voltage(self, index):
+        """Set the dac at the provided input to voltage output mode, powering down current mode
+
+        Args:
+            index (int): a number between zero and 3, telling us which dac we want to put into voltage mode
+        """
+        #There are only 4 dacs, so check there is a valid index
+        if (index >= 0 and index <= 3):
+            # Build the mask and value to write based on the index of the dac we want to enable. 
+            # To set a dac to voltage, we want to write 001 to the appropriate section on the COMMON_CONFIG register.
+            # The 00 sets the voltage output for the dac to powered up mode
+            # The 1 sets the current output for the dac to powered down mode
+            # See page 61 in the data sheet for more info on the COMMON_CONFIG register
+            mask = "0000" + ("000" * (3-index)) + "111" + ("000" * index)
+            write = "0000" + ("000" * (3-index)) + "001" + ("000" * index)
+            self.read_modify_write("COMMON_CONFIG", int(mask, 2), int(write, 2))
+        else:
+            print("Not a valid index.")
+            return
+        
+    def set_dac_current_range(self, current_range, index):
+        """specify a current range for the dac at the provided index 
+        
+        current_range (string):
+        "0000": 0 μA to 25 μA,
+        "0001": 0 μA to 50 μA,
+        "0010": 0 μA to 125 μA,
+        "0011": 0 μA to 250 μA,
+        "0100": 0 μA to -24 μA,
+        "0101": 0 μA to -48 μA,
+        "0110": 0 μA to -120 μA,
+        "0111": 0 μA to -240 μA,
+        "1000": -25 μA to +25 μA,
+        "1001": -50 μA to +50 μA,
+        "1010": -125 μA to +125 μA,
+        "1011": -250 μA to +250 μA,
+        index (int): the index of the dac we want to specify the current range for
+        """
+        #Change the current range for the DAC with the index provided. The value provided should be a binary string, one of those listed above. 
+        #See page 58 for more info on the DAC_X_IOUT_MISC_CONFIG register
+        self.read_modify_write("DAC_" + str(index) + "_IOUT_MISC_CONFIG", 0b0001111000000000, int(current_range + "000000000", 2))
+        
+    def set_dac_voltage(self, index, voltage, referenceVoltage):
+        """Set the output for the given index to the given voltage based on the reference voltage
+
+        Args:
+            index (int): the index of the dac we want to specify the voltage for
+            voltage (int): the voltage we want to set the output to
+            referenceVoltage (int): the reference voltage being used by the dac
+        """
+        value_to_write = voltage*4096/referenceVoltage 
```

Comment:
This also needs to include the current gain setting- therefore, you need to store the setting set in the  VOUT-GAIN-X register (either by reading it, or by storing the set value). You'll also need to know which reference is currently being used for this calculation- either the external value (user supplied and stored), internal (constant) or VDD (user supplied).